Remove unused code in settingsFragment

domdomegg opened the pull request:
Code is no longer required as entires and entryValues are already set in `preferences.xml`.

Want to tidy up SingleUploadFragment in a similar way, although it's not as simple as the Spinner class doesn't take an entryValues attribute.

Tested on APIs 15, 19, 23, 25 VMs and API 24 physical device (Samsung S6).

Previously was part of #455

whym commented:
Thanks for catching this. The redundant part was inserted by me in #509.

I believe the Travis CI failure can be fixed by modifying SettingsActivityTest.java: license_name_cc_by_4_0 -> license_name_cc_by_four (and maybe the corresponding comments above those lines).

whym commented:
I get crashes in non-English locales with these procedures:

* Change Android's language to "Deutch (Deutchland)" (German) and select "Attribution 4.0".
* Change Android's language to "日本語" (Japanese) and select "表示-継承 3.0".

The error I get is:

```
java.lang.RuntimeException: Unrecognized license value
at fr.free.nrw.commons.Utils.licenseNameFor(Utils.java:239)
at fr.free.nrw.commons.settings.SettingsFragment$1.onPreferenceChange(SettingsFragment.java:28)
at android.preference.Preference.callChangeListener(Preference.java:928)
at android.preference.ListPreference.onDialogClosed(ListPreference.java:281)
at android.preference.DialogPreference.onDismiss(DialogPreference.java:391)
at android.app.Dialog$ListenersHandler.handleMessage(Dialog.java:1263)
at android.os.Handler.dispatchMessage(Handler.java:102)
```

Does licenseNameFor perhaps use the wrong string IDs?

domdomegg commented:
Great spot that it breaks with certain locales on 3.0 licenses - can you double check that German broke on Attribution 4.0 because I can only get it to crash (with the same error) using version 3.0 licenses.

The reason this breaks is because the punctuation is slightly different - e.g. German locales use `CC-BY 3.0` whereas English it's `CC BY 3.0`. Therefore I've hardcoded the values in arrays.xml so they don't change around depending on locale, kind of the equivalent of Prefs (which hopefully we'll eventually transition away from).

domdomegg commented:
Please do test as thoroughly as you did before btw ;)

whym commented:
> can you double check that German broke on Attribution 4.0 because I can only get it to crash (with the same error) using version 3.0 licenses.

You are right, I must have mistyped it. I confirm it doesn't happen with 4.0 for the German locale.

It seems like there are two problems to solve.

1. The default license value is still localized (android:defaultValue="@string/license_name_cc_by_sa_4_0" in preferences.xml) and can cause a crash. Perhaps we'd want to create a new non-translatable string for this, and maybe for all the four license options for consistency. Procedure to reproduce the crash: Clear the app's data, change the locale to Portuguese and go to the app's Settings screen.
2. How to deal with the old license values already stored in the user's device. When the user is upgrading from an old version to this version, I believe the old stored will be incompatible and the app can only show the license screen with no item selected. Would that behavior be acceptable?

domdomegg commented:
For (1) I've added 5 untranslatable entries in strings for the prefs - as for (2) having tested it on German, French and English locales it seems to be happy going back and forwards from 2.2.2. Nevertheless, I've added a check in SingleUploadFragment that the spinner position is valid.

domdomegg commented:
New test output with changes ("All 2 tests passed"):

```
Testing started at 22:54 ...

05/10 22:54:30: Launching Tests in 'fr.free.nrw.commons'
$ adb push /home/[my files]/apps-android-commons/app/build/outputs/apk/app-debug.apk /data/local/tmp/fr.free.nrw.commons
$ adb shell pm install -r "/data/local/tmp/fr.free.nrw.commons"
Success


$ adb push /home/[my files]/apps-android-commons/app/build/outputs/apk/app-debug-androidTest.apk /data/local/tmp/fr.free.nrw.commons.test
$ adb shell pm install -r "/data/local/tmp/fr.free.nrw.commons.test"
Success


Running tests

$ adb shell am instrument -w -r   -e package fr.free.nrw.commons -e debug false fr.free.nrw.commons.test/android.support.test.runner.AndroidJUnitRunner
Client not ready yet..
Started running tests
Tests ran to completion.

```

domdomegg commented on app/src/androidTest/java/fr/free/nrw/commons/SettingsActivityTest.java:
```diff
@@ -15,15 +11,18 @@
+
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.anything;
+
```
Sorry about this unrelated change - Android Studio automatically reordered them without me noticing - can revert if needs be

domdomegg commented:
Fixed last commit, reverted moving imports around

whym commented:
Looks good, thanks. And thank you especially for taking care of the tests. :)

I tried it again, and my experience with the updated patch and v2.2.2 was that it goes back to the default CC BY-SA 4.0 from CC BY 3.0 after upgrade in the Japanese locale. Maybe environment dependent? (Mine was an API 22 emulator.)

Anyway, falling back from incompatible values to the default may be okay as long as the behavior is mentioned in the release notes.

whym commented:
Looking at https://github.com/wikimedia/mediawiki-extensions-Translate/blob/master/ffs/AndroidXmlFFS.php and https://github.com/wikimedia/translatewiki/blob/master/groups/Wikimedia/CommonsAndroid.yaml, it appears that "translatable" does not work with TranslateWiki.Net.

I'm not entirely confident, but I believe a workaround is moving them to a new resource file (e.g. res/values/keys.xml).

misaochan commented:
Sorry, just clarifying, is this in mergeable condition now? @whym would you like to merge?

whym commented:
Is there a need to merge this quickly? If that is the case, I can send a patch to implement my suggestion above later, hopefully soon enough.

whym commented:
I checked TranslateWiki.Net and the non-traslatable strings in the separate XML did not seem to go to there, which is good. :)
https://translatewiki.net/w/i.php?limit=50&title=Special%3AContributions&contribs=user&target=FuzzyBot&namespace=1206&tagfilter=&year=2017&month=-1
